wallet/src/states.rs

```diff
@@ -1,39 +1,22 @@
-use std::{
-    borrow::Borrow,
-    collections::HashMap,
-    hash::{Hash, Hasher},
-    iter::FusedIterator,
-};
+use std::{borrow::Borrow, hash::Hash};
+
+use hashlink::LinkedHashMap;
 
 #[derive(Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Hash, Debug)]
 pub enum Status {
     Confirmed,
     Pending,
 }
 
-struct State<K, S> {
+#[derive(Debug)]
+pub struct StateRef<K, S> {
     key: K,
```

Comment:
Why do we keep the key inside this struct?

Reply:
I'm actually not sure. It was already there and I didn't remove it.
But it's not necessary as far as I see.

Reply:
already removed this